core/repository/referredUsers/index.ts

```diff
@@ -0,0 +1,77 @@
+import config from 'shared/config';
+import { Account, Address } from 'shared/model/account';
+import Node, { FactorType } from 'core/repository/referredUsers/Node';
+
+import { ReferredUser } from 'core/repository/referredUsers/referredUser';
+
+class ReferredUsersRepository {
+
+    private addressToNode: Map<Address, Node>;
+
+    constructor() {
+        this.addressToNode = new Map();
+    }
+
+    getReferredUsers(account: Account, level: number) {
+        const node = this.getNode(account);
+
+        if (node !== undefined) {
+            return [...node.children.values()].map(item => new ReferredUser(item, level));
+        }
+
+        return [];
+    }
+
+    serialize(referredUser: ReferredUser): object {
```

Comment:
we can create NodeEntity class with fields only. Then extend NodeEntity with methods (`addFactor`, `addChild`, etc...). And we can use NodeEntity for typing.